setup.js

```diff
@@ -0,0 +1,184 @@
+//text to be displayed
+var displayText;
+var displayCommand;
+var displayNote;
+var displayVelocity;
+
+// Get viewport height, make canvas same dimension, for full screen.
+// Usually causes scrollbars.
+function windowResized() {
+  resizeCanvas(windowWidth, windowHeight);
+}
+
+function setup() {
+  createCanvas(windowWidth, windowHeight);
+
+  //starting background color
+  bgColor= color(220,220,200);
+
+  //starting square color
+  squareColor = color(100);
+
+  //Starting text
+  displayText="";
+  displayCommand="Command not received";
+  displayNote="Note not received";
+  displayVelocity="Velocity not received";
+
+  ////
+  //Setting up MIDI
+  ////
+	WebMidi.enable(function (err) { //check if WebMidi.js is enabled
+
+    // Request MIDI access
+    if (navigator.requestMIDIAccess) {
+        console.log('This browser supports WebMIDI!');
+
+        navigator.requestMIDIAccess().then(onMIDISuccess, onMIDIFailure);
+
+    } else {
+        console.log('WebMIDI is not supported in this browser.');
+    }
+    // Function to run when requestMIDIAccess is successful
+    function onMIDISuccess(midiAccess) {
+
+    //name our visible MIDI input and output ports
+        var inputs = midiAccess.inputs;
+        var outputs = midiAccess.outputs;
+
+        // Attach MIDI event "listeners" to each input
+        for (var input of midiAccess.inputs.values()) {
+            input.onmidimessage = getMIDIMessage;
+        }
+    }
+
+    // Function to run when requestMIDIAccess fails
+    function onMIDIFailure() {
+        console.log('Error: Could not access MIDI devices.');
+    }
+
+    //Choose an input port
+    inputSoftware = WebMidi.inputs[1];
```

Comment:
This can change depending on one's MIDI controller. I've used LoopMIDI and ableton, a Novation LaunchControl, LaunchControlXL, and LaunchKey. We should consider using the ID. A nice way to switch between controllers is the goal.